meson.build

```diff
@@ -0,0 +1,86 @@
+#                                                                                 #
+# NAME: meson.build                                                               #
+#                                                                                 #
+# AUTHOR: Michael Brockus.                                                        #
+#                                                                                 #
+# CONTACT: <mailto:michael@squidfarts.com>                                        #
+#                                                                                 #
+# NOTICES:                                                                        #
+#                                                                                 #
+# MIT License                                                                     #
+#                                                                                 #
+# Copyright (c) 2019 Team Squidfarts                                              #
+#                                                                                 #
+# Permission is hereby granted, free of charge, to any person obtaining a copy    #
+# of this software and associated documentation files (the "Software"), to deal   #
+# in the Software without restriction, including without limitation the rights    #
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell       #
+# copies of the Software, and to permit persons to whom the Software is           #
+# furnished to do so, subject to the following conditions:                        #
+#                                                                                 #
+# The above copyright notice and this permission notice shall be included in all  #
+# copies or substantial portions of the Software.                                 #
+#                                                                                 #
+# DISCLAIMERS:                                                                    #
+#                                                                                 #
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR      #
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,        #
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE     #
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER          #
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,   #
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE   #
+# SOFTWARE.                                                                       #
+#                                                                                 #
+###################################################################################
+
+
+
+project('facil.io', 'c',
+    version         : '0.2.4',
```

Comment:
Is this a facil.io version specifier? 

If so, could it be dynamically loaded? How?

Reply:
That is supposed to be the facil.io version number, if I was off then feel free to charge it to the correct number.  I don’t think there is a dynamic way to pre-load the version.

Reply:
Cool 👍🏻

FYI: the [current version is 0.7.2](https://github.com/boazsegev/facil.io/blob/fe7574d70c0bd3aa73a8d592cb902cfd6b8b8837/lib/facil/fio.h#L110-L113). In [0.8.x the version number will appear here](https://github.com/boazsegev/facil.io/blob/c35d19002c20f88696cb67224d4cb26a349290a5/lib/facil/fio-stl.h#L165-L168).

Reply:
> I don’t think there is a dynamic way to pre-load the version.

Hmm... not very comfortable... I wouldn't want to have the version number appear more than once, since it would be easy to make errors when maintaining the project.

Maybe we I could write a small script that will compile facil.io and bring the version to stdout, allowing us to "pipe" the version into the mason.build file during updates.

I'll have to think about this.

Reply:
Meson has a method for the project version number.  This can be done by calling ‘meson.project_verson()’.

Reply:
If the approach you're referencing is the one [suggested here](https://github.com/mesonbuild/meson/issues/4144), then I'm against it.

Using `meson.project_verson()` to export the version number to a compilation flag makes it hard to support semantical versioning and makes it impossible to support multiple build systems in a single project.

Both make and CMake are currently supported in addition to the future mason support. I won't make mason exclusive because it requires a version number.

Reply:
Version number is not required to define the project, I just added that so the user may use a wrap file to download the preferred version of facil.io.  Should I remove it?

Reply:
Maybe removing the version number would make more sense... yes.

I'm just not sure I understand, how does providing a version number help allows a developer to download the correct version? Right now, this is managed through git tags / releases. But I don't know that mason has a download manager...?

Reply:
Meson has this [WrapDB](https://mesonbuild.com/Wrap-dependency-system-manual.html) system for downloading third party dependencies.
With GitHub a git-wrap file would do the trick for automatically downloading 'facial.io'.

Reply:
Exposure of version is removed.

Reply:
Anything on that script yet?

Reply:
I'll write it later. Let's leave out the version for now and fix things up as we go.

Reply:
Version number removed.